src/ol/style/Style.js

```diff
@@ -7,6 +7,18 @@ import CircleStyle from '../style/Circle.js';
 import Fill from '../style/Fill.js';
 import Stroke from '../style/Stroke.js';
 
+
+/**
+ * A function that takes an {@link module:ol/Feature} and a `{number}`
+ * representing the view's resolution. The function should return a
+ * {@link module:ol/style/Style} or an array of them. This way e.g. a
+ * vector layer can be styled.
+ *
+ * @typedef {function((module:ol/Feature|module:ol/render/Feature), number):
+ *     (module:ol/style/Style|Array.<module:ol/style/Style>)} Function
```

Comment:
Does the Compiler care if we have a typedef named `Function` even though `Function` is [already a type](https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System).

Reply:
It works because typedefs need to be imported in the transformed code, so `Function` ends up being part of a variable name. But you're right, it looks dangerous so I'm going to rename it to `StyleFunction`. BTW, the reason why I moved this from `ol/style` to `ol/style/Style` is because the compiler did not find it there, for some reason I can not make sense of. I agree with what you said a few days ago: TypeScript is looking good right now.